welkin/models/base.py

```diff
@@ -20,11 +23,23 @@ def __getattr__(self, name):
 
 
 class Resource(dict, SchemaBase):
+    nested_objects = {}
+
     def __getattr__(self, name):
         try:
-            return super().__getitem__(name)
+            value = super().__getitem__(name)
         except KeyError:
-            return super().__getattr__(name)
+            value = super().__getattr__(name)
+
+        if name in self.nested_objects:
+            cls = getattr(sys.modules["welkin.models"], self.nested_objects[name])
+            if issubclass(cls, Collection):
+                obj = cls()
+                obj.extend([cls.resource(item) for item in value])
+                value = obj
+            else:
+                value = cls(value)
+        return value
```

Comment:
Not a requirement, but I prefer return statements (or blocks of return statements) be on their own line.

```python
if a:
    return True
else:
    return False

return None
```

I like to do this because when I'm browsing code, the first things I look for are inputs and outputs. If returns are in their own block, I don't have to hunt for them visually. I've found this pattern especially useful when a method has multiple return points.